routes/post.js

```diff
@@ -12,86 +12,122 @@ function resError(res, message){
-		content = content.replace(/(?:\r\n|\r|\n)/g, '<br />');
+	// Replace the newlines in the user's post with break tags in order to save the message 
+	// properly. This will allow us to display the post properly when the conversation page is 
+	// loaded again in situations where the user uses newlines in their post (such as when creating 
+	// lists or when creating spacing between typed paragraph). Note that we also need to replace 
+	// newlines with break tags in the display screen immediately after the user hits 'Send' on 
+	// their post. This is acheived by performing the same logic on the value of the text field when 
+	// #mainform is submited.
+	content = content.replace(/(?:\r\n|\r|\n)/g, '<br />');
 
-		// Create and save the post
-		var post = new db.models.Post({
-		    username	: req.body.username,
-		    userid		: req.body.userId,
-		    content  	: content,
-		    time  		: Date.now()
-		});
-		post.save(function(err, message){
-			if (err){
-				resError(res, "Could not save message into db.");
-			} else {
-				// Add the post to the conversation's discussion and save
-			    Conversation.findOne( { _id: req.body.conversationId }, function (err, convo, count){
-			    	if (err || !convo){
-						resError(res, "Could not find messages collection in DB.");
-					}
-					else{
-						convo.discussion.push(post._id);
-				    	convo.save(function(err, message){
-				    		if (err){
-								resError(res, "Could not save message into db.");
-							} else {
-								res.send({status: 'OK', success: true});
-							}
-						});
+	// Create and save the post. Note that the 'username' for a post is really 
+	// the pretty name formed from the user's first and last name.
+	var post = new db.models.Post({
+	    username	: name,
+	    userid		: userId,
+	    content  	: content,
+	    time  		: Date.now()
+	});
+	post.save(function(err, message){
+		if (err){
+			return onFailure("Could not save message into db.");
+		}
 
-						// Try to send emails to participants.
-						var participantIds = convo.participants.map(function (user){
-							return user._id;
-						});
-						var promise = User.find({ _id: { $in: participantIds }}).exec();
-						promise.then(function (participants){
-							var currentUser = participants.filter(function (user){
-								return user._id.toString() === req.body.userId;
-							})[0];
-							var emails = participants.map(function (user){return user.username});
-							var subject = 'GC: ' + currentUser.firstName + ' responded to the question: ' + convo.question;
-							if (subject.length > 70){
-								subject = subject.slice(0, 70) + '...';
-							}
-							var contentSnippet = content.slice(0, 140) + '...';
+		// Add the post to the conversation's discussion and save
+	    Conversation.findOne( { _id: conversationId }, function (err, convo, count){
+	    	if (err || !convo){
+	    		return onFailure("Could not find messages collection in DB.");
+			}
```

Comment:
This parenthesis can be moved back one tab, I think, to match up with the rest of the if statement.